react/src/views/Transfers/CreateTransferShipment/CreateTransferShipmentView.tsx

```diff
@@ -0,0 +1,178 @@
+import { useContext } from "react";
+import { gql, useMutation, useQuery } from "@apollo/client";
+import { Alert, AlertIcon, Center } from "@chakra-ui/react";
+import { useErrorHandling } from "hooks/error-handling";
+import { useNotification } from "hooks/hooks";
+import APILoadingIndicator from "components/APILoadingIndicator";
+import { useNavigate, useParams } from "react-router-dom";
+import { GlobalPreferencesContext } from "providers/GlobalPreferencesProvider";
+
+import {
+  AllAcceptedTransferAgreementsQuery,
+  CreateTransferShipmentMutation,
+  CreateTransferShipmentMutationVariables,
+  TransferAgreementType,
+} from "types/generated/graphql";
+import { BASE_ORG_FIELDS_FRAGMENT, TRANSFER_AGREEMENT_FIELDS_FRAGMENT } from "queries/fragments";
+import CreateTransferShipment, {
+  IOrganisationsAgreementsDataData,
+  ITransferShipmentFormData,
+} from "./components/CreateTransferShipment";
+
+export const ALL_ORGS_AND_BASES_WITH_TRANSFER_AGREEMENTS_QUERY = gql`
+  ${BASE_ORG_FIELDS_FRAGMENT}
+  ${TRANSFER_AGREEMENT_FIELDS_FRAGMENT}
+  query AllAcceptedTransferAgreements($baseId: ID!) {
+    base(id: $baseId) {
+      ...BaseOrgFields
+    }
+
+    transferAgreements(states: Accepted) {
+      ...TransferAgreementFields
+    }
+  }
+`;
+
+export const CREATE_SHIPMENT_MUTATION = gql`
+  mutation CreateTransferShipment(
+    $sourceBaseId: Int!
+    $targetBaseId: Int!
+    $transferAgreementId: Int!
+  ) {
+    createShipment(
+      creationInput: {
+        sourceBaseId: $sourceBaseId
+        targetBaseId: $targetBaseId
+        transferAgreementId: $transferAgreementId
+      }
+    ) {
+      id
+    }
+  }
+`;
+
+function CreateTransferShipmentView() {
+  // Basics
+  const navigate = useNavigate();
+  const { triggerError } = useErrorHandling();
+  const { createToast } = useNotification();
+  const { globalPreferences } = useContext(GlobalPreferencesContext);
+
+  // variables in URL
+  const baseId = useParams<{ baseId: string }>().baseId!;
+
+  // Query Data for the Form
+  const allFormOptions = useQuery<AllAcceptedTransferAgreementsQuery>(
+    ALL_ORGS_AND_BASES_WITH_TRANSFER_AGREEMENTS_QUERY,
+    {
+      variables: {
+        baseId,
+      },
+    },
+  );
+
+  // Mutation after form submission
+  const [createTransferShipmentMutation, createTransferShipmentMutationState] = useMutation<
+    CreateTransferShipmentMutation,
+    CreateTransferShipmentMutationVariables
+  >(CREATE_SHIPMENT_MUTATION);
+
+  // Prep data for Form
+  const allTransferAgreements = allFormOptions.data?.transferAgreements;
+
+  // eslint-disable-next-line max-len
+  const currentOrganisationLabel = `${allFormOptions?.data?.base?.organisation?.name} - ${allFormOptions?.data?.base?.name}`;
+
+  const partnerOrgsAgreementData = allTransferAgreements
```

Comment:
@vahidbazzaz can you explain what happens here? The list of all accepted agreements is filtered for these that have the user's org as source and that have types SendingTo or Bidirectional.
But it should be _all_ accepted agreements that have the user's org as source, independent of the type (shipments are always created by the source side).

Reply:
@vahidbazzaz added a (hopefully) clarifying note above :upside_down_face: